### PR TITLE
SCM-948 - Make limit parameter work for svnexe and gitexe changelog goal

### PR DESCRIPTION
**Pull request by cquoss**

### CODE REVIEW

**Comment by eolivelli** [comment]
Can you please explain more your patch?

**Comment by cquoss** [comment]
Happily. The old PR was better documented but got erased after resetting hard to upstream master and force pushing.
At our company we use a customized version of maven-release that talks in some added phases with our jira.
To be more precise we extract in one phase the current jira issue related with this maven project from the latest scm commit comment.
To not have the scm provider always drill down through the complete log history i also customized maven-scm in our nexus repository to only return the latest log entry by making the limit parameter functional for the changelog goal for the two providers we are interested in (gitexe and svnexe).
My patch here may not be perfect but with also the abstraction and the already deprecated methods i already made two attempts (the first inventing new methods and instantly deprecating them, the second removing all deprecation from the code and directly go through  request -> parameters to command line creation). 
Both have been rejected so now this is my third attempt doing this by overriding the createCommand method for these two providers mentioned.

**Comment by cquoss** [comment]
Another thing: Travis CI builds for jdk 7 and 8 have errors. To me it looks like jdk 7 and 8 are not available. Is some on it to fix the config on this?

File: maven-scm-providers/maven-scm-providers-git/maven-scm-provider-gitexe/src/main/java/org/apache/maven/scm/provider/git/gitexe/command/changelog/GitChangeLogCommand.java [hunk 1]
```diff
@@ -148,9 +169,21 @@ protected ChangeLogScmResult executeChangeLogCommand( ScmProviderRepository repo
     // ----------------------------------------------------------------------
 
     /**
-     * this constructs creates the commandline for the git-whatchanged command.
+     * <p>
```
Comment (michael-o) [inline maven-scm-providers/maven-scm-providers-git/maven-scm-provider-gitexe/src/main/java/org/apache/maven/scm/provider/git/gitexe/command/changelog/GitChangeLogCommand.java:172]: This `<p>` isn't necessary.

**Review by michael-o — CHANGES_REQUESTED** [review]

**Comment by michael-o** [comment]
Note: The order how those options are processed differ between Subversion and Git. This is bad for consistency, but not your fault I do not expect you to fix that. The change looks fine to me.

**Review by michael-o — APPROVED** [review]

**Pull request closed without merging**